cabal-install/Distribution/Client/Compat/FileLock.hsc

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE InterruptibleFFI #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module Distribution.Client.Compat.FileLock (
```

Comment:
I would love it if, for all future Compat modules we add, we also put an 'expiration date' on them; e.g., when they can be safely dropped.